### PR TITLE
 Adding programs into the services API 

### PR DESCRIPTION
**Pull request by joeldenning**
We should wait for Vicky, Luis, and Leonel to get back to us before merging this one. Once we merge it, it will run a data migration on the prod database with all the new programs and services.

### CODE REVIEW

File: migrations/sqls/20190726234757-programs-up.sql [hunk 1]
```diff
@@ -0,0 +1,155 @@
+/* Replace with your SQL commands */
```
Comment (joeldenning) [inline migrations/sqls/20190726234757-programs-up.sql]: This file is somewhat scary -- it's trying to reconcile the services that we have in the db already with the services that are in https://docs.google.com/document/d/1OgZf5acOTpbd-O0Up8ItRlW81Vspq8Rc6yrES1JlwLk/edit. Also it's creating a programs table and connecting all services to a program.

File: api-docs/list-services.md [hunk 2]
```diff
@@ -17,12 +17,26 @@ GET /api/services
     {
       "id": 1,
       "serviceName": "Citizenship",
-      "serviceDescription": "Gain United States citizenship"
+      "serviceDescription": "Gain United States citizenship",
+      "programId": 1,
+      "programName": "Immigration"
```
Comment (joeldenning) [inline api-docs/list-services.md]: This changes the `/api/services` api to give information about programs, too.

File: migrations/sqls/20190726234757-programs-up.sql [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  ("Family Support", "Family Support Services"),
+  ("Community Engagement and Organizing", "Community Engagement and Organizing")
+;
+
+ALTER TABLE services
+  ADD COLUMN programId INT,
+  ADD FOREIGN KEY programIdFk(programId) REFERENCES programs(id);
+
+UPDATE services SET programId = (
+  SELECT id from programs WHERE programName = "Preventive Health"
```
Comment (joeldenning) [inline migrations/sqls/20190726234757-programs-up.sql]: This should be Immigration

File: migrations/sqls/20190726234757-programs-up.sql [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  SELECT id from programs WHERE programName = "Youth Groups"
+) WHERE serviceName = "Youth Groups";
+
+UPDATE services SET programId = (
+  SELECT id from programs WHERE programName = "Nutrition / CRYS / SNAP"
+) WHERE serviceName = "SNAP";
+
+UPDATE services SET programId = (
+  SELECT id from programs WHERE programName = "Preventive Health"
+) WHERE serviceName = "Chronic Disease Testing";
```
Comment (MeghanBomberger) [inline migrations/sqls/20190726234757-programs-up.sql]: "Chronic Diseases Testing" doesn't appear to be on the intake form anymore. At least not the June 2019 copy.

File: migrations/sqls/20190726234757-programs-up.sql [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  SELECT id from programs WHERE programName = "Nutrition / CRYS / SNAP"
+) WHERE serviceName = "Grocery Store Tour";
+
+UPDATE services SET programId = (
+  SELECT id from programs WHERE programName = "Nutrition / CRYS / SNAP"
+) WHERE serviceName = "Cooking Classes";
+
+UPDATE services SET programId = (
+  SELECT id from programs WHERE programName = "Financial Education / Coaching"
+) WHERE serviceName = "Financial Coach";
```
Comment (MeghanBomberger) [inline migrations/sqls/20190726234757-programs-up.sql]: The June 2019 copy of the form seems to also have eliminated the separation between "Financial Coach" and "Financial Education".

File: migrations/sqls/20190726234757-programs-up.sql (hunk 4 above)
Comment (joeldenning) [reply migrations/sqls/20190726234757-programs-up.sql]: See https://docs.google.com/document/d/1fF-8KjJG16Yk7I_qG2X4FpydZRgFtM2yiyHmiLNjSjI/edit which is the most recent form that Vicky just sent me. The name has now changed to Chronic Diseases Screening, but it still exists.

File: migrations/sqls/20190726234757-programs-up.sql (hunk 5 above)
Comment (joeldenning) [reply migrations/sqls/20190726234757-programs-up.sql]: That google doc seems to sometimes have it split and sometimes not:

image

image

image

Vicky's version of the programs and services *does* have them split. See https://docs.google.com/document/d/1fF-8KjJG16Yk7I_qG2X4FpydZRgFtM2yiyHmiLNjSjI/edit

**Pull request merged**